src/components/search/filter/SearchFilterFacetPanels.vue

```diff
@@ -41,65 +56,56 @@ export default class SearchFilterFacetPanels extends Vue {
     };
   }
 
-  private topFieldSelector(top: string[], type: string): ISearchFacetField[] {
-    const fields = [] as ISearchFacetField[];
-    for (const value of top) {
-      const field = this.fieldSelector(type).find(
-        (field) => this.fieldValueFormatter(field.value) === value
-      );
-      fields.push({
-        value: field?.value ? this.fieldValueFormatter(field?.value) : value,
-        count: field?.count || 0,
-        text: field?.text || "",
-      });
-    }
-    return fields;
-  }
-
-  get topEntityTypes(): ISearchFacetField[] {
-    return this.topFieldSelector(this.topTypes, "entity_type");
-  }
-
-  get moreEntityTypes(): ISearchFacetField[] {
-    return this.fieldSelector("entity_type")
-      .filter(
-        (field) =>
-          !this.topTypes.includes(this.fieldValueFormatter(field.value))
-      )
-      .map((field) => ({
-        value: this.fieldValueFormatter(field?.value) || "",
-        count: field?.count || 0,
-        text: field?.text || "",
-      }));
-  }
-
-  get topEntityStatuses(): ISearchFacetField[] {
-    return this.topFieldSelector(this.topStatuses, "entity_status");
-  }
-
-  get facets(): ISearchFacetRecord {
-    return this.searchTopicFacets;
-  }
-
-  get fields(): Record<string, ISearchFacetField[] | unknown> {
-    return (this?.facets?.fields || {}) as Record<
-      string,
-      ISearchFacetField[] | unknown
-    >;
+  get topEntityTypes(): ISearchFilter[] {
+    return topFieldSelector(
+      {
+        ...entityTypeSpec,
+        inclusions: this.topTypes,
+        keySelector: (filter?: ISearchFilter) =>
+          fieldKeyFormatter((filter?.value || "::") as string),
+        valueSelector: (filter?: ISearchFilter) =>
+          fieldValueFormatter((filter?.value || "::") as string),
+      },
+      this.searchFilterFields.category as unknown as ISearchFilter[]
+    );
   }
 
-  get categories(): ISearchFacetField[] {
-    return (this?.fields?.category || []) as ISearchFacetField[];
+  get moreEntityTypes(): ISearchFilter[] {
+    return moreFieldSelector(
+      {
+        ...entityTypeSpec,
+        exclusions: this.topTypes,
+        keySelector: (filter?: ISearchFilter) =>
+          fieldKeyFormatter((filter?.value || "::") as string),
+        valueSelector: (filter?: ISearchFilter) =>
+          fieldValueFormatter((filter?.value || "::") as string),
+      },
+      this.searchFilterFields.category as unknown as ISearchFilter[]
+    );
   }
 
-  fieldSelector(type: string): ISearchFacetField[] {
-    return (this.categories as ISearchFacetField[]).filter((cat) =>
-      (cat.value as string).includes(type)
-    );
+  get entityStatuses(): ISearchFilter[] {
+    const options = {
+      ...entityStatusSpec,
```

Comment:
one thing we may want to consider {...object} is only a shallow copy, not a deep copy. We may want to look at doing deep copies using lodash-es for better encapsulation

Reply:
True, but the spec as designed is only ever one level deep